ui/src/components/SampleQueue/CurrentTree.jsx

```diff
@@ -59,173 +59,165 @@ export default class CurrentTree extends React.Component {
-                    shapes={this.props.shapes}
-                    showDialog={this.props.showDialog}
-                    showWorkflowParametersDialog={
-                      this.props.showWorkflowParametersDialog
-                    }
-                  />
-                );
-
-                break;
-              }
-              case 'xrf_spectrum': {
-                task = (
-                  <XRFTaskItem
-                    key={taskData.queueID}
-                    index={i}
-                    id={`${taskData.queueID}`}
-                    data={taskData}
-                    sampleId={sampleData.sampleID}
-                    checked={this.props.checked}
-                    showForm={this.props.showForm}
-                    plotsData={this.props.plotsData}
-                    plotsInfo={this.props.plotsInfo}
-                    showDialog={this.props.showDialog}
-                  />
-                );
-
-                break;
-              }
-              case 'energy_scan': {
-                task = (
-                  <EnergyScanTaskItem
-                    key={taskData.queueID}
-                    index={i}
-                    id={`${taskData.queueID}`}
-                    data={taskData}
-                    sampleId={sampleData.sampleID}
-                    checked={this.props.checked}
-                    showForm={this.props.showForm}
-                    shapes={this.props.shapes}
-                    showDialog={this.props.showDialog}
-                  />
-                );
-
-                break;
-              }
-              case 'Characterisation': {
-                task = (
-                  <CharacterisationTaskItem
-                    key={taskData.queueID}
-                    index={i}
-                    id={`${taskData.queueID}`}
-                    data={taskData}
-                    sampleId={sampleData.sampleID}
-                    checked={this.props.checked}
-                    state={
-                      this.props.sampleList[taskData.sampleID].tasks[i].state
-                    }
-                    showForm={this.props.showForm}
-                    addTask={this.props.addTask}
-                    shapes={this.props.shapes}
-                    showDialog={this.props.showDialog}
-                  />
-                );
-
-                break;
-              }
-              default: {
-                task = (
-                  <TaskItem
-                    key={taskData.queueID}
-                    index={i}
-                    id={`${taskData.queueID}`}
-                    data={taskData}
-                    sampleId={sampleData.sampleID}
-                    checked={this.props.checked}
-                    showForm={this.props.showForm}
-                    shapes={this.props.shapes}
-                    showDialog={this.props.showDialog}
-                  />
-                );
-              }
-            }
+  if (sampleId) {
```

Comment:
I guess this in principle could go below the `if` but It does not seem like it was like that before

Reply:
Ha, yes. I'm getting rid of the `show` prop anyway in #1984